### PR TITLE
Allow timeout/conectionTimeout for requests

### LINKED ISSUE #86: Support timeout inside connection config
If we request and need a response from InfluxDB but we may want to have a timeout to stop InfluxDB request instead letting them running indefinitely.

It would be great to be able to set a timeout directly inside Influx connection config, for example : 

    influx_db:
        connections:
            default:
                host: '%influxdb_host%'
                database: '%influxdb_db%'
                username: '%influxdb_user%'
                password: '%influxdb_password%'
                timeout: 2

### PR DESCRIPTION
**Pull request by Jerome1337**
Fixes #86

### CODE REVIEW

**Comment by Algatux** [comment]
Hi @Jerome1337! Thx for your work, I will check and merge it Asap

File: README.MD [hunk 1]
```diff
@@ -61,6 +61,10 @@ influx_db:
             http_port:            8086
             username:             ''
             password:             ''
+
+            # Setup timeout or connection timeout (seconds) for your requests
+            timeout:              0
```
Comment (Algatux) [inline README.MD]: please specify .0 to be more strict on the type

File: README.MD [hunk 2]
```diff
@@ -79,6 +83,10 @@ influx_db:
     http_port:            8086
     username:             ''
     password:             ''
+
+    # Setup timeout or connection timeout (seconds) for your requests
+    timeout:              0
```
Comment (Algatux) [inline README.MD]: same here

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -62,6 +62,8 @@ public function getConfigTreeBuilder()
                             ->integerNode('http_port')->defaultValue(8086)->end()
                             ->scalarNode('username')->defaultValue('')->end()
                             ->scalarNode('password')->defaultValue('')->end()
+                            ->floatNode('timeout')->defaultValue(0)->end()
```
Comment (Algatux) [inline src/DependencyInjection/Configuration.php]: here too -> .0

File: src/Services/ConnectionFactory.php [hunk 4]
```diff
@@ -40,7 +42,9 @@ public function createConnection(
         int $udpPort,
         string $user,
         string $password,
-        bool $udp = false
+        bool $udp = false,
+        float $timeout = 0,
```
Comment (Algatux) [inline src/Services/ConnectionFactory.php]: default must be .0

File: src/Services/ConnectionFactory.php [hunk 5]
```diff
@@ -71,9 +77,11 @@ private function createClient(
...
```
Comment (Algatux) [inline src/Services/ConnectionFactory.php]: here too :P

**Review by Algatux — CHANGES_REQUESTED** [review]
I know that the values from configuration comes as they are but in future I want to enforce the strict typings on them, so please always specify the correct value type

File: README.MD (hunk 1 above)
Comment (soullivaneuh) [reply README.MD]: Why? Does Guzzle accept float? :thinking:

Comment (soullivaneuh) [reply README.MD]: Forget my comment: http://docs.guzzlephp.org/en/stable/request-options.html#timeout

File: README.MD [hunk 6]
```diff
@@ -67,6 +67,10 @@ influx_db:
             http_port:            8086
             username:             ''
             password:             ''
+
+            # Setup timeout or connection timeout (seconds) for your requests
```
Comment (soullivaneuh) [inline README.MD:71]: This should be also on the `info()` method of the configuration.

Please add this. The best thing to do for the documentation is to use `bin/console config:dump-reference` and copy/paste the result.

**Review by soullivaneuh — CHANGES_REQUESTED** [review]

**Comment by Jerome1337** [comment]
@Algatux @Soullivaneuh I fixed your requested changes.

**Comment by soullivaneuh** [comment]
Thank you @Jerome1337

**Comment by soullivaneuh** [comment]
I'll make a release for this and the SSL feature @Algatux :+1:

**Pull request merged**